R/utils_render_table_codecheckers.R

```diff
@@ -0,0 +1,79 @@
+#' Renders codecheckers table in JSON format.
+#' 
+#' @param list_codechecker_reg_tables The list of codechecker register tables. The indices are the ORCID IDs.
+render_table_codecheckers_json <- function(list_codechecker_reg_tables){
+  list_orcid_ids <- names(list_codechecker_reg_tables)
+  # Check.names arg is set to FALSE so that column "ORCID ID" has the space
+  table_codecheckers <- data.frame(`ORCID ID`= list_orcid_ids, stringsAsFactors = FALSE, check.names = FALSE)
+  
+  # Column- codechecker names
+  table_codecheckers$`Codechecker name` <- sapply(
+    X = table_codecheckers$`ORCID ID`,
+    FUN = function(orcid_id) {
+      paste0(CONFIG$DICT_ORCID_ID_NAME[orcid_id])
+    }
+  )
+
+  # Column- No. of codechecks
+  table_codecheckers$`No. of codechecks` <- sapply(
+    X = table_codecheckers$`ORCID ID`,
+    FUN = function(orcid_id) {
+      paste0(nrow(list_codechecker_reg_tables[[orcid_id]]))
+    }
+  )
+
+  # Arranging the column names
+  table_codecheckers <- table_codecheckers[, c("Codechecker name", "ORCID ID", "No. of codechecks")]
+
+  return(table_codecheckers)
+}
+
+#' Renders codecheckers table in HTML format.
+#' Each codechecker name links to the register table for that specific
+#' codechecker. The ORCID IDs link to their ORCID pages.
+#' 
+#' @param list_codechecker_reg_tables The list of codechecker register tables. The indices are the ORCID IDs.
+render_table_codecheckers_html <- function(list_codechecker_reg_tables){
+
+  list_orcid_ids <- names(list_codechecker_reg_tables)
+  table_codecheckers <- data.frame(ORCID_ID = list_orcid_ids, stringsAsFactors = FALSE)
+
+  # Column- codechecker names
+  # Each codechecker name will be a hyperlink to the register table
+  # with all their codechecks
+  table_codecheckers$`Codechecker name` <- sapply(
+    X = table_codecheckers$ORCID_ID,
+    FUN = function(orcid_id) {
+      codechecker_name <- CONFIG$DICT_ORCID_ID_NAME[orcid_id]
+      paste0("[", codechecker_name, "](https://codecheck.org.uk/register/codecheckers/",
+      orcid_id, "/)")
+    }
+  )
+
+  # Column- ORCID ID link
+  table_codecheckers$ORCID_ID_Link <- sapply(
+    X = table_codecheckers$ORCID_ID,
+    FUN = function(orcid_id) {
+      paste0("[", orcid_id, "](https://orcid.org/", orcid_id, ")")
+    }
+  )
+  # Column- No. of codechecks
+  # Shown as "no_codechecks (sell all checks)" where "see all checks" links to the checks by 
+  # the codechecker
+  table_codecheckers$`No. of codechecks` <- sapply(
+    X = table_codecheckers$ORCID_ID,
+    FUN = function(orcid_id) {
+      no_codechecks <- nrow(list_codechecker_reg_tables[[orcid_id]])
+      paste0(nrow(list_codechecker_reg_tables[[orcid_id]])," [(see all checks)](https://codecheck.org.uk/register/codecheckers/",
+      orcid_id, "/)")
+    }
+  )
+
+  # Drop the original ORCID_ID column and rename ORCID_ID_Link to ORCID_ID
+  table_codecheckers <- table_codecheckers[, c("Codechecker name", "ORCID_ID_Link", "No. of codechecks")]
```

Comment:
This is just one of many lines where the column name is used. I would prefer it if the name would only be configured once at the top of this file, in case the names should be changed. For example, I am unsure how common names with spaces are in a JSON file...

You could replace the `$` operator with `[[` in the other lines, I think:

```r
  table_codecheckers[["No. of codechecks"]] <- sapply(
```

An then you could reuse a variable. Can you please try that?

Reply:
Good point, thanks. I created this list
`CONFIG$CODECHECKER_TABLE_COL_NAMES <- list(
  "codechecker" = "Codechecker name",
  "orcid" = "ORCID ID",
  "no_codechecks" = "No. of codechecks"
)
`

and the function now retrieves the column names from this list.
I realized I should make similar changes for the other non_register table render functions. I'll create a separate PR for that.